sdk/openai/Azure.AI.OpenAI/src/Custom/StreamingCompletions.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Azure.Core.Sse;
+
+namespace Azure.AI.OpenAI.Custom
+{
+    public class StreamingCompletions : IDisposable
+    {
+        private Response _baseResponse;
+        private bool _disposedValue;
+        private SseReader _reader;
+
+        internal StreamingCompletions(Response response)
+        {
+            _baseResponse = response;
+            _reader = new SseReader(response.ContentStream);
+        }
+
+        public async IAsyncEnumerable<string> GetAsyncMessages()
```

Comment:
Async should be a suffix. It's a .NET idiom.

Reply:
Thanks -- this method isn't intended to actually exist/persist beyond the first data flow demonstration, but that's no excuse for even a code-sharing draft PR to flaunt proper patterns.